examples/live/check_results.py

```diff
@@ -6,98 +6,202 @@
-                    or psd_epoch < sim_gps_start or psd_epoch > sim_gps_end:
-                log.info('Invalid PSD in %s %s', trigfp, detector)
-                fail = True
-
-            if 'snr' not in group or len(group['snr']) == 0:
-                continue
-
-            detectors_with_trigs.add(detector)
-
-            # check that SNR is non-negative and finite
-            snr = group['snr'][:]
-            if (snr < 0).any() or not np.isfinite(snr).all():
-                log.error('Invalid SNR in %s %s', trigfp, detector)
-                fail = True
-
-            # check that Allen chi^2 is non-negative and finite
-            chisq = group['chisq'][:]
-            chisq_dof = group['chisq_dof'][:]
-            if (chisq < 0).any() or not np.isfinite(chisq).all() \
-                    or (chisq_dof < 0).any() or not np.isfinite(chisq_dof).all():
-                log.error('Invalid Allen chi^2 in %s %s', trigfp, detector)
-                fail = True
-
-            # check that merger time is within the simulated time range
-            end_time = group['end_time'][:]
-            if (end_time < sim_gps_start).any() or (end_time > sim_gps_end).any():
-                log.error('Invalid merger time in %s %s', trigfp, detector)
-                fail = True
-
-            # check that template parameters are consistent with the bank
-            trig_mass1 = group['mass1'][:]
-            trig_mass2 = group['mass2'][:]
-            trig_s1z = group['spin1z'][:]
-            trig_s2z = group['spin2z'][:]
-            trig_temp_id = group['template_id'][:]
-            test_mass1 = close(temp_mass1[trig_temp_id], trig_mass1)
-            test_mass2 = close(temp_mass2[trig_temp_id], trig_mass2)
-            test_s1z = close(temp_s1z[trig_temp_id], trig_s1z)
-            test_s2z = close(temp_s2z[trig_temp_id], trig_s2z)
-            test_all = np.logical_and.reduce((test_mass1, test_mass2,
-                                              test_s1z, test_s2z))
-            if not test_all.all():
-                log.error('Invalid template parameters in %s %s',
-                              trigfp, detector)
-                fail = True
-
-# check that triggers were produced in all detectors
-if detectors_with_trigs != tested_detectors:
-    missing = sorted(tested_detectors - detectors_with_trigs)
-    log.error('No triggers found in %s', ', '.join(missing))
-    fail = True
-
+lsctables.use_in(LIGOLWContentHandler)
+      
+inj_xmls = sorted(glob.glob('./hwinjcbc*.xml.gz'))
+if len(inj_xmls)==0:
+      log.error('No inj_xml found when checking results')
+      fail=True
+else:
+      path_to_inj_xml = inj_xmls[0]
+      log.info('injection xml found')
+      inj_doc = ligolw_utils.load_filename(
+            path_to_inj_xml, False, contenthandler=LIGOLWContentHandler)
+      inj_table = lsctables.SnglInspiralTable.get_table(inj_doc)
+   
+      tested_detectors = get_tested_detectors(inj_table)
```

Comment:
I see the reasoning behind this, but inferring the list of detectors from the injections is not going to be obvious. I suggest keeping the original definition of `tested_detectors`. The point is that check_results.py has to be consistent with whatever set of detectors we decided to simulate, regardless of the injections, and I think it is ok to hardcode this both in `check_results.py` and `run.sh`.

Reply:
This was prep for the possibility of not always using HLV (I've been talking with Peter about doing a quick investigation into the effect of filtering out noise around 60 Hz on the SNR we pick up in H and L). My idea is that at some point in the future, the code will first check what detectors the injection is specified for, then only simulate strain for those detectors.

Reply:
Well, let's not mix other investigations into this, this script is strictly to provide a rapid sanity check of PyCBC Live and a simple example. I think hardcoding to HLV is ok for the foreseeable future of this script. It would of course be good to test other detector combinations in a more thorough test, but I doubt we want to run such a long test on Travis at every commit.

Reply:
Ok I will remove this from here. I'm probably going to wind up building a more complicated test system at some point but ya it probably won't be for travis.